magicgui/types.py

```diff
@@ -24,6 +24,9 @@
 #: A function that takes a ``(value, annotation)`` argument and returns an optional
 #: :attr:`WidgetTuple`
 TypeMatcher = Callable[[Any, Optional[Type]], Optional[WidgetTuple]]
+#: A function that takes a ``(value, annotation)`` argument and returns an optional
+#: :attr:`WidgetTuple`
+ReturnMatcher = Callable[[Any, Optional[Type]], Optional[WidgetTuple]]
```

Comment:
since this is just an alias (not an actual class or anything that could be used for `isinstance` for example) and since this alias is exactly the same as `TypeMatcher`.  Let's just use `TypeMatcher` for both.